knode/api.go

```diff
@@ -57,14 +58,21 @@ func NewPrivateValidatorAPI(kcoin *Kowala) *PrivateValidatorAPI {
 }
 
 // Start the validator.
-func (api *PrivateValidatorAPI) Start() error {
+func (api *PrivateValidatorAPI) Start(deposit *big.Int) error {
```

Comment:
do we really need this, it add a dependency on validator

Reply:
I don't think we need it.
But if we keep this argument here, shall we get rid of the `setDeposit` and make the `deposit` mandatory?

Reply:
It's an optional parameter. A user can start a validator with `validator.setDeposit(10); validator.start();` or `validator.start(10);`

Reply:
@acroca I like the idea of removing setDeposit. It can prevent a lot of errors.

Reply:
:+1:

Reply:
@acroca @jmatosp removed. I kept `validator.start([deposit])` with an optional parameter to handle scenario:
1. start with deposit 10
2. stop
3. start without the deposit parameter to use old value.